src/game/moveFinding/findBishopBaseMoves.js

```diff
@@ -1,3 +1,97 @@
-export default function findBishopBaseMoves() {
-    return [];
+
+import BoardCoordinates from "../board/boardCoordinates";
+
+export default function findBishopBaseMoves(boardSituation, startingCoordinates, color) {
+    let currentBoard = boardSituation.position;
+    return checkMoves(currentBoard, startingCoordinates, color);
+
+}
+
+function checkMoves(currentBoard, startingCoordinates, color) {
+
+    let posibleMoves = []; //Array with all possible moves
+    let current = new BoardCoordinates(startingCoordinates.x, startingCoordinates.y);
+
+    //Change code to similar like rook piece 
+
+    //4 loop while for check every diagonal 
+    // on the next revision try to do with one function
+    let x = startingCoordinates.x;
+    let y = startingCoordinates.y;
+    x++;
+    y++;
+    while (BoardCoordinates.valid(current)) {
```

Comment:
To będzie nieskończona pętla (jeśli jej break nie zatrzyma) bo inkrementujesz x oraz y a sprawdzasz current który nigdy nie jest mutowany.

Reply:
Za dużo wzorowania na kodzie Jacka... chciałem wykorzystać funkcję w klasie BoardCoordinates, tak jak pisałeś. Zaraz spróbuje to jakoś sensownie poprawić. Zostanie oczywiście jeszcze sprawa redundacji kodu.